napari/_qt/menus/plugins_menu.py

```diff
@@ -113,9 +117,22 @@ def _add_toggle_widget(*, key=key, hook_type=hook_type):
                 menu.addAction(action)
             action.triggered.connect(_add_toggle_widget)
 
+    def _plugin_manager_dialog_cls(self) -> Union[type, None]:
+        """Return the plugin manager class, if available."""
+        try:
+            # TODO: Register via plugin system?
```

Comment:
Could you expand on what this means? Thanks

Reply:
Using the [entry point](https://napari.org/stable/plugins/manifest.html) to register the plugin manager itself as a plugin too, like other plugins do. Right now it's not technically a plugin, but an external dependency that we enable on conditionally depending on its availability at import time.

Reply:
Thanks! And would the plugin manager plugin just contribute one command that returns `QtPluginDialog` ? (Would it be a [`Widget`](https://napari.org/dev/plugins/contributions.html#contributions-widgets) command?)

I guess when plugin[ menu contributions](https://github.com/orgs/napari/projects/24/views/2?pane=issue&itemId=30502168) eventually gets implemented, plugin manager could directly contribute the `_show_plugin_install_dialog` menu item but that may be a while away.

Reply:
>  Right now it's not technically a plugin,

Thanks, that part was confusing as we call it a plugin atm

Reply:
Yea my personal preference would be to wait for a menu contribution plugin directly, instead of using existing approaches that wouldn't fit the use case perfectly.